htlcswitch/test_utils.go

```diff
@@ -1345,15 +1345,13 @@ func (n *twoHopNetwork) makeHoldPayment(sendingPeer, receivingPeer lnpeer.Peer,
 	}
 
 	// Send payment and expose err channel.
-	go func() {
-		err := sender.htlcSwitch.SendHTLC(
-			firstHop, pid, htlc,
-		)
-		if err != nil {
-			paymentErr <- err
-			return
-		}
+	err = sender.htlcSwitch.SendHTLC(firstHop, pid, htlc)
+	if err != nil {
+		paymentErr <- err
+		return paymentErr
+	}
```

Comment:
Does the caller see the difference here? Or do we want the channel read to certainly succeed if there was a problem during init of the htlc?

Reply:
yes, with this approach one could do a non-blocking select immediately after the call returns to see if it failed before being added to the switch. currently it is racey because it is done from a go routine, so the caller would have to guess some sort of timeout before concluding the htlc was indeed added